packages/tgc/src/common.components/search-input/search-input.tsx

```diff
@@ -0,0 +1,137 @@
+import { getUserTypeSetting } from '../../store/slices/settings';
+import {
+    clearSuggestions, clearTwitchUser,
+    getIsSuggestionsLoading,
+    getIsTwitchUserFetching,
+    getSuggestions,
+} from '../../store/slices/twitch-user';
+import { getIsAuth } from '../../store/slices/user';
+import { addNotification, isValidSearchChange, isValidSearchSubmit } from '../../utils';
+
+import SearchInputSuggestions from './__suggestions/search-input__suggestions';
+
+const SearchInput = () => {
+    const dispatch = useDispatch();
+    const [usernameParam] = useState('');
+    const isAuth = useSelector(getIsAuth);
+
+    const intl = useIntl();
+    const [location, navigate] = useLocation();
+
+    const suggestions = useSelector(getSuggestions);
+    const isSuggestionsLoading = useSelector(getIsSuggestionsLoading);
+
+    const isSearchingUser = useSelector(getIsTwitchUserFetching);
+
+    const [typingTimer, setTypingTimer] = useState<NodeJS.Timeout | null>(null);
+    const [username, setUsername] = useState(usernameParam || '');
+    const userType = useSelector(getUserTypeSetting);
+
+    const [fetchSuggestions] = useLazyGetDisplayNameSuggestionsQuery();
+
+    useEffect(() => {
+        if (!isValidSearchSubmit(userType, username)) {
+            setUsername('');
+        }
+    }, [userType]);
+
+    useEffect(() => {
+        if (usernameParam) {
+            setUsername(usernameParam);
+        }
+    }, [usernameParam]);
+
+    const handleSubmit = (displayName: string | null = null) => {
+        const searchName = displayName || username;
+
+        if (typingTimer) {
+            clearTimeout(typingTimer);
+        }
+
+        if (!isValidSearchSubmit(userType, searchName)) {
+            addNotification({
+                id: `notification.searchInput.${userType}.submit`,
+                type: SNACKBAR_TYPE.ERROR,
+                autoHideDuration: NOTIFICATIONS_DURATION.S,
+            }, dispatch);
+            return;
+        }
+
+        if (searchName === usernameParam) {
+            setUsername(searchName);
+            return;
+        }
+
+        dispatch(clearTwitchUser());
+        dispatch(clearChannelsState());
+        dispatch(clearMessages());
+        dispatch(clearSuggestions());
+
+        navigate(`${LINKS.MESSAGES}/${searchName}`);
+    };
+    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        if (!isValidSearchChange(userType, e.target.value)) {
+            addNotification({
+                id: `notification.searchInput.${userType}`,
+                type: SNACKBAR_TYPE.WARNING,
+                autoHideDuration: NOTIFICATIONS_DURATION.S,
+            }, dispatch);
+            return;
+        }
+
+        if (e.target.value.length <= 2) {
+            dispatch(clearSuggestions());
+        }
+
+        setUsername(e.target.value);
+    };
+
+    const handleSelect = (displayName: string) => {
+        handleSubmit(displayName);
```

Comment:
as we should remove displayName from handleSubmit function, we just need firstly update our displayName variable and then call handleSubmit